src/bounded_vec.rs

```diff
@@ -436,6 +438,157 @@ impl<T, const L: usize, const U: usize> OptBoundedVecToVec<T> for Option<Bounded
     }
 }
 
+/// Suports encoding and decoding with [borsh](https://crates.io/crates/borsh), and BorsSchema.
+///
+/// By default Borsh uses u32 as length prefix for sequences.
+/// For bounded we used u8, u16 or u32 depending on the U.
+/// Increase or decreaasing U may not always be backward compatible.
+#[cfg(feature = "borsh")]
+mod borsh_impl {
+    use super::*;
+
+    use alloc::collections::btree_map::{BTreeMap, Entry};
+    use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+
+    impl<T: BorshSerialize, const L: usize, const U: usize> BorshSerialize for BoundedVec<T, L, U> {
+        fn serialize<W: borsh::io::Write>(&self, writer: &mut W) -> borsh::io::Result<()> {
+            if U <= u8::MAX as usize {
+                (self.len() as u8).serialize(writer)?;
+            } else if U <= u16::MAX as usize {
+                (self.len() as u16).serialize(writer)?;
+            } else {
+                (self.len() as u32).serialize(writer)?;
```

Comment:
Borsh seems to do a fallible usize->u32 conversion when serializing slices

Reply:
Yes, will do. Will wait for json serde merge first, it will conflict with this PR.

Reply:
Seems I fixed. Approach is to panic (more idiomatic for bounded vector) with some strife for constantness.

Reply:
- not used as - as does not panics nor in runtime nor in compile. in runtime even with overflow flag
- used from were it works, used expect when cannot prove compile time (so it is kind of design time failure)

imho best approach. closing please comment if other ideas.